iOS Release Notes June 2021 (#2984)
Azure/azure-sdk: releases/2021-06/ios.md

To add the Azure SDK for iOS to your application, follow the instructions in [Adding Package Dependencies to Your App](https://developer.apple.com/documentation/xcode/adding_package_dependencies_to_your_app):

> With your project open in Xcode 11 or later, select **File > Swift Packages > Add Package Dependency...** Enter the clone URL of this repository: *https://github.com/Azure/azure-sdk-for-ios.git* and click **Next**. For the version rule, specify the exact version or version range you wish to use with your application and click **Next**. Finally, place a checkmark next to each client library you wish to use with your application, ensure your application target is selected in the **Add to target** dropdown, and click **Finish**.

Author: Created a separate PR for this.

### Swift CLI

To add the Azure SDK for iOS to your application, follow the example in [Importing Dependencies](https://swift.org/package-manager/#importing-dependencies):

> Open your project's `Package.swift` file and add a new package dependency to your project's `dependencies` section, specifying the clone URL of the Swift Package Manager mirror repository and the version specifier you wish to use:

Reviewer: Same comment about adding this to the template.

```swift
// swift-tools-version:5.3
    dependencies: [
        ...
        .package(name: "AzureCore", url: "https://github.com/Azure/SwiftPM-AzureCore.git", from: "1.0.0-beta.12")

>     ],

Author: Fixed.

>     ],

Reviewer: Are we purposefully leaving Calling out?

Author: Yes. Calling does not support SwiftPM

```

Next, add each client library you wish to use in a target to the target's array of `dependencies`:

```swift
    targets: [
        ...
        .target(
            name: "MyTarget",
            dependencies: ["AzureCore", ...])
    ]

> )

Author: Fixed.

> )

Reviewer: Are we purposefully leaving Calling out?

Author: Yes

## Release highlights

### Azure Communication Services Calling

#### 1.1.0-beta.1(https://github.com/Azure/azure-sdk-for-ios/blob/master/sdk/communication/AzureCommunicationCalling/CHANGELOG.md#110-beta1-2021-06-04)

##### New features
- Support for CallKit (**Preview mode**)
  - Use the api `createCallAgentWithCallKitOptions` to create `CallAgent` with `CallKit` enabled and SDK will report to `CallKit` about incoming call , outgoing calls and all other call operations like `mute`, `unmute`, `hold`, `resume` as part of the API calls. 
  - When the app is in the killed state and incoming call is received use the api `reportToCallKit`.

> 

Reviewer: nit;
```suggestion
```

- `CallAgent` and `CallClient` now has `dispose` API to explicitly delete the objects instead of relying on ARC.

> 

Reviewer: nit;
```suggestion
```

> - Get CorrelationId from `CallInfo` object in `Call` to get the id required for recording feature. 

Reviewer: ```suggestion
- Get `CorrelationId` from the `CallInfo` object in a `Call` to get the id required for the recording feature.
```

> - Support to start recording by an ACS endpoint.

Reviewer: ```suggestion
- Support for starting a recording by an ACS endpoint.
```

##### Bug fixes
- [iOS] ACSRendererView layout is off after a device rotation https://github.com/Azure/Communication/issues/127.
- [iOS] Resizing issue for animating streams https://github.com/Azure/Communication/issues/262.

> - Creating multiple CallAgents with same token will throw error.

Author: I just copied what Calling had put in their release notes, but I rephrased these as you suggested.

### Azure Communication Services Chat

#### 1.0.0-beta.12(https://github.com/Azure/azure-sdk-for-ios/blob/master/sdk/communication/AzureCommunicationChat/CHANGELOG.md#100-beta12-2021-06-07)

##### Breaking Changes
- Changed the way in which options are instantiated for the following classes: `CreateChatThreadOptions`, `DeleteChatThreadOptions`,  `ListChatThreadsOptions`, `AddChatParticipantsOptions`, `DeleteChatMessageOptions`, `GetChatMessageOptions`, `GetChatThreadPropertiesOptions`, `ListChatMessagesOptions`, `ListChatParticipantsOptions`, `ListChatReadReceiptsOptions`, `RemoveChatParticipantOptions`, `SendChatMessageOptions`, `SendChatReadReceiptOptions`, `SendTypingNotificationOptions`, `UpdateChatMessageOptions`, `UpdateChatThreadPropertiesOptions`.
    - old:
        `let options = Chat.CreatChatThreadOptions()`
    - new:  
        `let options = CreateChatThreadOptions()`
- Moved `AzureCommunicationChatClient.ApiVersion` to `AzureCommunicationChatClientOptions.ApiVersion`.

> - Renamed `CommunicationError` to `ChatError`

Reviewer: nit;
```suggestion
- Renamed `CommunicationError` to `ChatError`.
```

- Removed following classes:  `CreateChatThreadResult`, `CreateChatThreadRequest`, `ChatMessage`, `ChatMessageContent`, `ChatParticipant`, `ChatMessageReadReceipt`, `ChatThreadProperties`.

> - Removed Any type in TrouterEventUtil, and create a new enum TrouterEvent

Reviewer: nit;
```suggestion
- Removed Any type in TrouterEventUtil, and create a new enum TrouterEvent.
```
